### PR TITLE
change default in ds.chunk and datarray.chunk variable.chunk

### PR DESCRIPTION
**Pull request by aurghs**
The aim is to split the PR #4595 in small PRs.
The scope of this smaller PR is to modify the default of `chunks` in `dataset.chunk` to align the behaviour to `xr.open_dataset`. 
The main changes are:
- Modify the default of chunk in `dataset.chunk`, `datarray.chunk` and `variable.chunk` from None to {}.
- If the user pass `chunk=None` inside is set to `{}`
- Add a future warning to advice that the usage of `None` will raise an error in the future. 

Note that the changes currently don't modify the behaviour of `dataset.chunk`

 - [x] Related #4496
 - [x] Passes `isort . && black . && mypy . && flake8`

### CODE REVIEW

**Review by alexamici — APPROVED** [review]
This is the agreed change to the `.chunks` default value.

There is no user change now, as `None` is still accepted, but deprecated to avoid that the meaning is different from `chunks=None` in `xr.open_dataset`.

File: xarray/core/dataarray.py [hunk 1]
```diff
@@ -1010,12 +1010,11 @@ def chunks(self) -> Optional[Tuple[Tuple[int, ...], ...]]:
     def chunk(
         self,
         chunks: Union[
-            None,
             Number,
             Tuple[Number, ...],
             Tuple[Tuple[Number, ...], ...],
             Mapping[Hashable, Union[None, Number, Tuple[Number, ...]]],
-        ] = None,
+        ] = {},
```
Comment (shoyer) [inline xarray/core/dataarray.py:1017]: can we add a comment that `{}` is being used intentionally here, even though it's technically unsafe?

**Review by shoyer — COMMENTED** [review]
cc @jhamman for a second look -- we talked about this a few weeks ago

File: xarray/core/dataset.py [hunk 2]
```diff
@@ -1855,17 +1853,22 @@ def chunk(
...
-                    "object: %s" % bad_dims
-                )
+        bad_dims = chunks.keys() - self.dims.keys()
+        if bad_dims:
+            raise ValueError(
+                "some chunks keys are not dimensions on this " "object: %s" % bad_dims
+            )
 
         variables = {
             k: _maybe_chunk(k, v, chunks, token, lock, name_prefix)
```
Comment (jhamman) [inline xarray/core/dataset.py:1874]: The style checker is catching that `name_prefix` is not defined here.

**Comment by jhamman** [comment]
Other than a small comment above, this looks good to me.

File: xarray/core/dataset.py (hunk 2 above)
Comment (aurghs) [reply xarray/core/dataset.py:1874]: Yes, sorry, I deleted a line by mistake yesterday, when I have added the inline comment -_-'

**Pull request merged**